Auto-Deployment.md

```diff
@@ -0,0 +1,72 @@
+## To set up an auto-deploy project:
+
+1. Create the actual App Engine project (or switch to an existing one) using `gcloud init`
+
+2. Enable
+    [_App Engine Admin_](https://pantheon.corp.google.com/apis/library/appengine.googleapis.com),
+    [_Cloud Build_](https://console.developers.google.com/apis/library/cloudkms.googleapis.com),
+    and
+    [_Cloud KMS_](https://console.developers.google.com/apis/library/cloudbuild.googleapis.com)
+    APIs
+
+3. Create the [Cloud KMS keyring](https://cloud.google.com/cloud-build/docs/securing-builds/use-encrypted-secrets-credentials#example_build_request_using_an_encrypted_variable)
+    - `gcloud kms keyrings create amp-github-apps-keyring --location=global`
+    > Note: Because sharing a keyring across projects is complex, we use the same keyring name (`amp-github-apps-keyring`) for each project keyring; this uniformity make s it simpler to use shared encrypt/decrypt scripts. We will likewise use the standard `app-env-key` as the key name.
+
+4. Provide team access to the keyring
+      ```
+      gcloud kms keyrings add-iam-policy-binding amp-github-apps-keyring \
+        --location=global \
+        --member group:amp-core-infra@google.com \
+        --role roles/cloudkms.cryptoKeyDecrypter
+      ```
+      > This will allow all members of the group to encrypt and decrypt secrets using this keyring.
+
+5. Create the Cloud KMS CryptoKey
+      ```
+      gcloud kms keys create app-env-key \
+        --location=global \
+        --keyring=amp-github-apps-keyring \
+        --purpose=encryption
+      ```
+
+6. Grant decryption permissions to the project service account
+    ```
+    gcloud kms keys add-iam-policy-binding app-env-key \
+      --location=global \
+      --keyring=amp-github-apps-keyring \
+      --member=serviceAccount:[PROJECT_NUMBER]@cloudbuild.gserviceaccount.com \
+      --role=roles/cloudkms.cryptoKeyDecrypter
```

Comment:
I considered adding a script that automatically does all this key setup by pulling from `gcloud info`. What do y'all think about that?

Reply:
No objection if it makes the default setup easier. However, if it adds an unnecessary step, maybe leave things as they are?